### DISCUSSION TITLE
Application is unavailable after version upgrade to 0.65.2

### DISCUSSION THREAD

**Discussion by dheeban-k** [outline/outline#3966] [Self Hosting]
[scroll.zingworks.com-1660424875409.log](https://github.com/outline/outline/files/9332178/scroll.zingworks.com-1660424875409.log)
<img width="1792" alt="Screenshot 2022-08-14 at 2 38 12 AM" src="https://user-images.githubusercontent.com/101546615/184510831-814745be-7754-4615-bcbd-9a8b5b4adef6.png">

Hi All,

My Entire production setup of outline is down. With the above attached error message and the browser logs. Please help me on this!

My app runs in Google Cloud run environment as docker image
I've exposed my variabled in .env file as environment variables and mounted secrets in Google secret manager
version upgraded to 0.65.2

**Comment by tommoor** [comment]
You don't mention what version you upgraded from, and what your upgrade process was. Most likely you just need to run the migrations

**Reply by dheeban-k** [reply]
I upgraded from 0.63.0 to 0.65.2 directly. db:migrate was successful too. Now I tried to go one by one version, that also failed and throwing error. @tommoor

**Reply by dheeban-k** [reply]
My Upgrade process involved only 3 steps:

docker run --env-file=.env --rm outlinewiki/outline yarn sequelize db:create
docker tag e42e3f2e6479 "google container registry path"/getoutline:v0.0.1
docker push "google container registry path"getoutline:v0.0.1

And finally , I deploy the IMAGE from GCR @tommoor

**Reply by tommoor** [reply]
So you didn't run db:migrate according to those steps?

**Reply by dheeban-k** [reply]
Oops...i ran db:migrate...missed to add in the above list.
When I ran this command docker run --env-file=.env --rm outlinewiki/outline yarn sequelize db:create the output was "Loaded configuration file "server/config/database.json".
Using environment "production".

ERROR: database "scroll-zingworks-p01" already exists"

And after running the same with db:migrate, several migrations happened.
Now when I re-run it, I'm getting "No migrations were executed, database schema was already up to date.
Done in 1.18s." @tommoor

**Reply by dheeban-k** [reply]
```
# –––––––––––––––– REQUIRED ––––––––––––––––

# Generate a hex-encoded 32-byte random key. You should use `openssl rand -hex 32`
# in your terminal to generate a random value.
SECRET_KEY=secret

# Generate a unique random key. The format is not important but you could still use
# `openssl rand -hex 32` in your terminal to produce this.
UTILS_SECRET=secret
# For production point these at your databases, in development the default
# should work out of the box.
DATABASE_URL=postgres://username:password@IP:5432/DB_NAME

# Uncomment this to disable SSL for connecting to Postgres
# PGSSLMODE=disable

# For redis you can either specify an ioredis compatible url like this
REDIS_URL=redis://IP:6379
# or alternatively, if you would like to provide addtional connection options,
# use a base64 encoded JSON connection option object. Refer to the ioredis documentation
# for a list of available options.
# Example: Use Redis Sentinel for high availability
# {"sentinels":[{"host":"sentinel-0","port":26379},{"host":"sentinel-1","port":26379}],"name":"mymaster"}
# REDIS_URL=ioredis://eyJzZW50aW5lbHMiOlt7Imhvc3QiOiJzZW50aW5lbC0wIiwicG9ydCI6MjYzNzl9LHsiaG9zdCI6InNlbnRpbmVsLTEiLCJwb3J0IjoyNjM3OX1dLCJuYW1lIjoibXltYXN0ZXIifQ==

# URL should point to the fully qualified, publicly accessible URL. If using a
# proxy the port in URL and PORT may be different.
URL=my endpoint
PORT=3000

# See [documentation](docs/SERVICES.md) on running a separate collaboration
# server, for normal operation this does not need to be set.
#COLLABORATION_URL=

# To support uploading of images for avatars and document attachments an
# s3-compatible storage must be provided. AWS S3 is recommended for redundency
# however if you want to keep all file storage local an alternative such as
# minio (https://github.com/minio/minio) can be used.

# A more detailed guide on setting up S3 is available here:
# => https://wiki.generaloutline.com/share/125de1cc-9ff6-424b-8415-0d58c809a40f
#
AWS_ACCESS_KEY_ID=ID
AWS_SECRET_ACCESS_KEY=Secret
AWS_REGION=us
AWS_S3_UPLOAD_BUCKET_URL=bucket url
AWS_S3_UPLOAD_BUCKET_NAME=bucket name
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private


# –––––––––––––– AUTHENTICATION ––––––––––––––

# Third party signin credentials, at least ONE OF EITHER Google, Slack,
# or Microsoft is required for a working installation or you'll have no sign-in
# options.


# To configure Google auth, you'll need to create an OAuth Client ID at
# => https://console.cloud.google.com/apis/credentials
#
# When configuring the Client ID, add an Authorized redirect URI:
# https://<URL>/auth/google.callback
GOOGLE_CLIENT_ID=client ID
GOOGLE_CLIENT_SECRET=clinet secret


# –––––––––––––––– OPTIONAL ––––––––––––––––


# Auto-redirect to https in production. The default is true but you may set to
# false if you can be sure that SSL is terminated at an external loadbalancer.
FORCE_HTTPS=true

# Have the installation check for updates by sending anonymized statistics to
# the maintainers
ENABLE_UPDATES=true

# How many processes should be spawned. As a reasonable rule divide your servers
# available memory by 512 for a rough estimate
WEB_CONCURRENCY=1

# Override the maxium size of document imports, could be required if you have
# especially large Word documents with embedded imagery
MAXIMUM_IMPORT_SIZE=5120000

# You can remove this line if your reverse proxy already logs incoming http
# requests and this ends up being duplicative
DEBUG=http

# Comma separated list of domains to be allowed to signin to the wiki. If not
# set, all domains are allowed by default when using Google OAuth to signin
ALLOWED_DOMAINS=our company domain
```

**Reply by dheeban-k** [reply]
Adding my .env file here @tommoor

**Reply by tommoor** [reply]
Feels like the `20220430043135-collection-sort-backfill` failed or did not run for some reason, you can check the database contents and see what the contents of the `collections.sort` column looks like. It should be filled by default with `{ field: "title", direction: "asc" }` for every collection.

**Reply by dheeban-k** [reply]
postgres=> SELECT * FROM collections.sort;
ERROR:  relation "collections.sort" does not exist
LINE 1: SELECT * FROM collections.sort;

@tommoor

**Reply by tommoor** [reply]
Then the migrations did not run or failed for some reason, look in the `SequelizeMeta` table to see which migrations have been ran. You could remove the row that relates to adding the collection sort column and then it will run again on next `db:migrate` call.

**Reply by dheeban-k** [reply]
okay will do that and get back to @tommoor . Thanks for your help so far !

**Reply by dheeban-k** [reply]
Hi @tommoor I re-ran the same steps and could now see the migration is successful and 0.65.2 is rocking !! Thanks a ton for ur support :)

**Comment by dheeban-k** [comment]
I upgraded from 0.63.0 to 0.65.2 directly. db:migrate was successful too. Now I tried to go one by one version, that also failed and throwing error. @tommoor
